tests/RobotFramework/tests/cumulocity/registration/device_registration.robot

```diff
@@ -193,13 +198,55 @@ Entities persisted and restored
         Should Have MQTT Messages    c8y/s/us/plc2    message_contains=102    date_from=${timestamp}    minimum=0    maximum=0
     END
 
+Entities send to cloud on restart
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/' '{"@type":"child-device","@id":"plc1"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/' '{"@type":"child-device","@id":"plc2"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/sensor1' '{"@type":"child-device","@id":"plc1-sensor1","@parent":"factory/shop/plc1/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/sensor2' '{"@type":"child-device","@id":"plc1-sensor2","@parent":"factory/shop/plc1/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/sensor1' '{"@type":"child-device","@id":"plc2-sensor1","@parent":"factory/shop/plc2/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/metrics' '{"@type":"service","@id":"plc1-metrics","@parent":"factory/shop/plc1/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/metrics' '{"@type":"service","@id":"plc2-metrics","@parent":"factory/shop/plc2/"}'
+
+    External Identity Should Exist    plc1
+    External Identity Should Exist    plc2
+    External Identity Should Exist    plc1-sensor1
+    External Identity Should Exist    plc1-sensor2
+    External Identity Should Exist    plc2-sensor1
+    External Identity Should Exist    plc1-metrics
+    External Identity Should Exist    plc2-metrics
+
+    Execute Command    cat /etc/tedge/.tedge-mapper-c8y/entity_store.jsonl
+    ${original_last_modified_time}=    Execute Command    date -r /etc/tedge/.tedge-mapper-c8y/entity_store.jsonl
+
+    FOR    ${counter}    IN RANGE    0    5
+        ${timestamp}=    Get Unix Timestamp
+        Restart Service    tedge-mapper-c8y
+        Service Health Status Should Be Up    tedge-mapper-c8y
+
+        # Assert that the file contents changed on restart
+        ${last_modified_time}=    Execute Command    date -r /etc/tedge/.tedge-mapper-c8y/entity_store.jsonl
+        Should Not Be Equal    ${last_modified_time}    ${original_last_modified_time}
```

Comment:
Since the timestamp assertion is already done in the previous test, we can either skip this whole block or at least avoid the looping and limit the check to only once.